cabal-install/Distribution/Client/List.hs

```diff
@@ -131,6 +134,13 @@ getPkgList verbosity packageDBs repoCtxt comp progdb listFlags pats = do
     return matches
   where
     onlyInstalled = fromFlag (listInstalled listFlags)
+    exactMatch = fromFlag (listExactMatch listFlags)
+    filterPackages :: [(PackageName, a, b)] -> [(PackageName, a, b)]
+    filterPackages =
+      filter
```

Comment:
Don't filter, rather define

```haskell
ipiSearch | exactMatch = InstalledPackageIndex.searchByNameExact
          | otherwise  = InstalledPackageIndex.searchByNameSubstring
```

and similarly for PackageIndex variant.

Note: `PackageIndex.searchByName` uses case-insensitive matching. I think we want exact, therefore need to do special variant. (Should be easy).